### PR TITLE
[feat]: [CDS-77315]: Converted null values to empty for Grpc call.

### PR DESCRIPTION
**Pull request by senjucanon2**
## Describe your changes:
Issue: For EditGitDetails feature, in case of account level templates, org and projId were sent as Null and are acceptable as values for a scope. Bur there was internal Grpc call which threw NPE as null is not supports null values for any fields in Grpc object.

Change: the null values for OrgId and ProjId are passed as emptyStrings during Grpc Call and at ScmService End they are set again to null.

Impact: Edit GitDetails feature.

Testing: Create Flow and editGitDetails flow have been tested with gitRepoAllowlist for templates at account level.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51897)

### CODE REVIEW

**Comment by senjucanon2** [comment]
trigger messagecheck

**Comment by senjucanon2** [comment]
trigger uts

**Review by adivishy1 — APPROVED** [review]
> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51897)* status: 5 file(s) to review. 1 of 2 approvals obtained.

File: 953-git-sync-commons/src/main/proto/io/harness/gitsync/harness_to_git.proto [hunk 1]
```diff
@@ -366,7 +366,7 @@ message GitSettings {
 }
 
 message ValidateRepoRequest {
-  ScopeIdentifiers scope = 1;
+  ScopeIdentifiers scopeIdentifiers = 1;
```
Comment (rohitk-h) [inline 953-git-sync-commons/src/main/proto/io/harness/gitsync/harness_to_git.proto:369]: Any specific reason to rename?

File: 900-git-sync-sdk/src/main/java/io/harness/gitsync/scm/SCMGitSyncHelper.java [hunk 2]
```diff
@@ -334,14 +333,15 @@ public ScmGetBatchFilesResponse getBatchFilesByBranch(
 
   public void validateRepo(
```
Comment (rohitk-h) [inline 900-git-sync-sdk/src/main/java/io/harness/gitsync/scm/SCMGitSyncHelper.java:334]: Can we add a unit test for the method?

File: 953-git-sync-commons/src/main/proto/io/harness/gitsync/harness_to_git.proto (hunk 1 above)
Comment (senjucanon2) [reply 953-git-sync-commons/src/main/proto/io/harness/gitsync/harness_to_git.proto:369]: Earlier the name of the variable was not in sync with the other objects declared in the class, I changed the name just to be in sync with other methods which are already present.

File: 900-git-sync-sdk/src/main/java/io/harness/gitsync/scm/SCMGitSyncHelper.java (hunk 2 above)
Comment (senjucanon2) [reply 900-git-sync-sdk/src/main/java/io/harness/gitsync/scm/SCMGitSyncHelper.java:334]: Added a UT

**Review by rohitk-h — APPROVED** [review]

**Comment by senjucanon2** [comment]
trigger uts

**Comment by mohitg0795** [comment]
*[`953-git-sync-commons/src/main/proto/io/harness/gitsync/harness_to_git.proto` line 369 at r1](https://reviewable.io/reviews/harness/harness-core/51897#gh-1302480096:-NcaC8-l6QChX13rwjeK:b-1z9bpv) ([raw file](https://github.com/harness/harness-core/blob/20f263d67dbbb1c3913060021b6a52d4c1a46d84/953-git-sync-commons/src/main/proto/io/harness/gitsync/harness_to_git.proto#L369)):*
<details><summary><i>Previously, senjucanon2 (Vivek Dixit) wrote…</i></summary><blockquote>

Earlier the name of the variable was not in sync with the other objects declared in the class, I changed the name just to be in sync with other methods which are already present.
</blockquote></details>

I think this is a backward incompatible change @senjucanon2

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51897)* status: 6 file(s) to review. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`953-git-sync-commons/src/main/proto/io/harness/gitsync/harness_to_git.proto` line 369 at r1](https://reviewable.io/reviews/harness/harness-core/51897#gh-1302480096:-NcayYnJ95rbZM7i3lk9:b-490ii7) ([raw file](https://github.com/harness/harness-core/blob/20f263d67dbbb1c3913060021b6a52d4c1a46d84/953-git-sync-commons/src/main/proto/io/harness/gitsync/harness_to_git.proto#L369)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

I think this is a backward incompatible change @senjucanon2
</blockquote></details>

Have reverted the change of name.

**Comment by senjucanon2** [comment]
trigger uts

**Review by mohitg0795 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51897)* status: 6 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Comment by senjucanon2** [comment]
trigger uts

**Review by adivishy1 — APPROVED** [review]
> Reviewed 1 of 1 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51897)* status: 6 file(s) to review. 2. unresolved discussions.

**Comment by senjucanon2** [comment]
trigger review

**Pull request merged**